### ISSUE TITLE
Ability to pass just a single object

### ISSUE THREAD

**Issue by balupton** [hyperhype/hyperscript#65]
So I coded up this file recently which uses hyperscript:
https://github.com/bevry/outpatient/blob/455fd3a32791259931e0fa4e3b9b3b7d1dc26d8b/source/block.js

The current argument reception means that I need to pass 3 different arguments, which results in code like this:

``` javascript
h('a', {
	class: 'prev',
	href: prev.url
}, [
	h('span', { class: 'icon' }),
	h('span', { class: 'title' }, prev.title)
])
```

It would be nice if I could just do this:

``` javascript
h({
	tag: 'a',
	class: 'prev',
	href: prev.url,
	children: [
		h('span', { class: 'icon' }),
		h('span', { class: 'title' }, prev.title)
	]
})
```

As when bundled together with tens of other elements, it will become a lot easier to read for multiline components.

**Comment by reqshark** [comment]
i agree, both ways of writing are good. while i might tend to the former, others may prefer the latter

**Comment by balupton** [comment]
so I did up a gist over at: https://gist.github.com/balupton/c807a9be215723f0e849697097b99a50

I'm not actually sure this suggestion results in better code, it sure makes understanding the hierarchy easier, but it does double the lines of code

**Comment by reqshark** [comment]
that gist makes things look more readable using the `children` style in my opinion

**Comment by dominictarr** [comment]
I could tolerate (what this will do to the implementation internals) if this only applies when the object is the first arg (if the tag name is not the string) and if all fields are under properties.

`` js
h({tag: ..., classes: ..., attributes: {...}, events: {...}, children: [...] })
```
with only a fixed number of accepted properties on the top object. then the code for handling any of those things can be refactored into a function and just called on the object.

I disagree that it's more "readable" but that is very subjective.

**Comment by reqshark** [comment]
yes of course, so it may not be more "readable" 

but there is something less subjective here if we believe in the following premise:

* a function that accepts fewer arguments is simpler

**Comment by reqshark** [comment]
also if we do `tag` and `children`, let's also do a textNode prop like  

```js
h({tag: ..., classes: ..., attributes: {...}, text: ... })
```

**Comment by dominictarr** [comment]
@ahdinosaur I am looking for the issue that you opened regards consolidating the hyperscript api, but can't find it.

**Comment by ahdinosaur** [comment]
@dominictarr thanks for the ping: https://github.com/hyperhype/hyperscript/issues/66

**Comment by dmitriz** [comment]
Passing objects is also easier from the Functional Programming perspective, if you want to compose `h` with some pure functions:

```js
var link = {tag: 'a', href: 'http://check.it', children:['Check my link!']}
var applyStyle = style => elm => {...elm, style}
var styledNode = h(applyStyle(link)) 
  //=> h({tag: 'a', href: 'http://check.it', style: style, children:['Check my link!']})
```

would be easier than

```js
var textInput = {tag: 'input, href: 'http://check.it', children:['Check my link!']}
var applyStyle = style => elm => {...elm, style}
var removeProps = props => obj => {
   ... // write function removing the props from the object
}
var toArray = obj => [obj.tag, obj, obj.children]
var styledNode = h.apply(this, toArray(removeProps(['tag', 'children'])(applyStyle(link))))
  //=> h('input', {type: 'text', style: style})
```

that you need now with the current implementation of `h`.
